java/Introduction/CompareBinaryToHex/CompareBinaryToHex.java

```diff
@@ -16,13 +16,12 @@ public static int digitToValue(char c) {
 	public static int convertToBase(String number, int base) {
 		if (base < 2 || (base > 10 && base != 16)) return -1;
 		int value = 0;
-		for (int i = number.length() - 1; i >= 0; i--) {

```

Comment:
For loop wasn't invoking String.length() at each iteration since we only read that value on the initialization part, which is done only once regardless of the amount of loops to be done.